### DISCUSSION TITLE
Why doesn't the JDK's thread pool plugin create a span when submitting or executing tasks?

### DISCUSSION THREAD

**Discussion by xiaqi1210** [apache/skywalking#12529] [Q&A] [answered]
Why doesn't the JDK's thread pool plugin create a span when submitting or executing tasks, but only when the task object actually starts running? 

This way, if I want to monitor the time spent from task submission to when the task starts running, it won't be possible.

Are there other considerations here, or would creating a span during `submit` or `execute` cause any issues?

**Comment by wu-sheng** [comment]
Are you activating boostrap plugin? I believe it is different from that.

https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/bootstrap-plugins/

**Comment by wu-sheng** [comment]
We only created one span in executing context to make sure the tracing continues.

**Reply by xiaqi1210** [reply]
if I want to monitor the time spent from task submission to when the task starts running, I create a span during submit or execute, such as the following figure.
![image](https://github.com/user-attachments/assets/b7f05568-bfa7-4c41-b12b-6c3af457b63f)

Is there any problem with this?

**Reply by wu-sheng** [reply, marked as answer]
If you want to create more local spans, you are free to change.
Make sure they are stopped correctly and note, more spans means more memory, network, and storage volume cost.
